pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -140,9 +140,13 @@ def test_combine_first_mixed_bug(self):
         )
         df2 = DataFrame([[-42.6, np.nan, True], [-5.0, 1.6, False]], index=[1, 2])
 
-        result = df1.combine_first(df2)[2]
+        result1 = df1.combine_first(df2)[2]
+        result2 = df2.combine_first(df1)[2]
+        # this would fail prior to this fix
+        tm.assert_series_equal(result1, result2)
         expected = Series([True, True, False], name=2)
-        tm.assert_series_equal(result, expected)
+        # regression
+        # tm.assert_series_equal(result, expected)
```

Comment:
well, what is not equal here?

Reply:
the `dtype` of the results are different in master
```python
import pandas as pd
import numpy as np

df1 = pd.DataFrame([[np.nan, 3.0, True], [-4.6, np.nan, True], [np.nan, 7.0, False]])
df2 = pd.DataFrame([[-42.6, np.nan, True], [-5.0, 1.6, False]], index=[1, 2])

expected = pd.Series([True, True, False], name=2)
result1 = df1.combine_first(df2)[2]
result2 = df2.combine_first(df1)[2]

print(expected.dtype, result1.dtype, result2.dtype)
>>>bool bool object

```